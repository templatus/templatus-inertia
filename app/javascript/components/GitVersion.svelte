<script lang="ts">
  import { onMount } from 'svelte';
  import * as timeago from 'timeago.js';

  let timeElement: HTMLElement;

  onMount(() => {
    timeago.render(timeElement);

    return () => timeago.cancel(timeElement);
  });

  export let commitVersion: string;
  export let commitTime: string;
</script>

<div>
  <code class="text-sm">{commitVersion}</code>
  <b class="ml-3"><time bind:this={timeElement} datetime={commitTime} /></b>
</div>
