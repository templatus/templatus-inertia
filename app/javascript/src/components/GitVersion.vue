<script setup lang="ts">
import { ref, onMounted, onBeforeUnmount } from 'vue';
import * as timeago from 'timeago.js';

defineProps({
  commitVersion: {
    type: String,
    required: true,
  },

  commitTime: {
    type: String,
    required: true,
  },
});

const timeElement = ref();

onMounted(() => {
  timeago.render(timeElement.value);
});

onBeforeUnmount(() => {
  timeago.cancel();
});
</script>

<template>
  <div>
    <code class="text-sm">{{ commitVersion }}</code>
    <b class="ml-3"><time ref="timeElement" :datetime="commitTime"></time></b>
  </div>
</template>
