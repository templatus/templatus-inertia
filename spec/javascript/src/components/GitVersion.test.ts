import { mount } from '@vue/test-utils';
import GitVersion from '@/components/GitVersion.vue';

describe('GitVersion', () => {
  test('matches snapshot', () => {
    const wrapper = mount(GitVersion, {
      props: {
        commitVersion: 'v0.0.1-123-7654321',
        commitTime: '2021-06-01T12:00:00+02:00',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();

    wrapper.unmount();
  });
});
