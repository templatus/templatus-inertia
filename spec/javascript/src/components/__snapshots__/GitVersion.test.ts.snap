// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitVersion matches snapshot 1`] = `<div><code class="text-sm">v0.0.1-123-7654321</code><b class="ml-3"><time datetime="2021-06-01T12:00:00+02:00" timeago-id="4"></time></b></div>`;
